addons/mail/static/src/components/file_uploader/file_uploader.js

```diff
@@ -97,14 +97,25 @@ class FileUploader extends Component {
      */
     async _performUpload(files) {
         for (const file of files) {
-            const response = await this.env.browser.fetch('/web/binary/upload_attachment', {
-                method: 'POST',
-                body: this._createFormData(file),
-            });
-            let html = await response.text();
-            const template = document.createElement('template');
-            template.innerHTML = html.trim();
-            window.eval(template.content.firstChild.textContent);
+            const tempAttachment = this.env.models['mail.attachment'].find(attachment =>
+                attachment.filename === file.name);
```

Comment:
Use `uploading` in the name, and reliance on just `filename === name` is quite weak, so also use `isTemporary` in find.